src/test/scala/uk/gov/nationalarchives/consignmentexport/GraphQlApiSpec.scala

```diff
@@ -165,4 +171,47 @@ class GraphQlApiSpec extends ExportSpec {
     }
     exception.getMessage should equal(s"No data returned from the update consignment status call for consignment $consignmentId ")
   }
+
+  "the getCustomMetadata method" should "return custom metadata" in {
+    val consignmentClient = mock[GraphQLClient[gce.Data, gce.Variables]]
+    val updateExportClient = mock[GraphQLClient[ued.Data, ued.Variables]]
+    val updateConsignmentStatus = mock[GraphQLClient[ucs.Data, ucs.Variables]]
+    val customMetadataClient: GraphQLClient[cm.Data, cm.Variables] = mock[GraphQLClient[cm.Data, cm.Variables]]
+    val keycloak = mock[KeycloakUtils]
+
+    doAnswer(() => Future(new BearerAccessToken("token"))).when(keycloak).serviceAccountToken[Identity](any[String], any[String])(
+      any[SttpBackend[Identity, Any]], any[ClassTag[Identity[_]]], any[TdrKeycloakDeployment])
+    doAnswer(() => Future(GraphQlResponse[cm.Data](Option(cm.Data(customMetadata)), Nil))).when(customMetadataClient)
+      .getResult[Identity](any[BearerAccessToken], any[Document], any[Option[cm.Variables]])(any[SttpBackend[Identity, Any]], any[ClassTag[Identity[_]]])
+
+
+    val consignmentId = UUID.randomUUID()
+    val api = new GraphQlApi(config, consignmentId, keycloak, consignmentClient, updateExportClient, updateConsignmentStatus, customMetadataClient)
+
+    val customMetadataResponse = api.getCustomMetadata.unsafeRunSync()
+    customMetadataResponse should equal(customMetadata)
+  }
+
+  "the getCustomMetadata method" should "return an error if there is no custom metadata" in {
```

Comment:
Is that true? Is there a legitimate case where there is no metadata, ie the user has not added any descriptive or closure metadata?

Reply:
I was wrong in what I said before, this is if there are no custom metadata definitions, so nothing in the FileProperty table. I don't know if that's likely enough to warrant a test though, what do you think?

Reply:
Reckon its worth keeping in, just renaming the test to reflect the meaning.

Reply:
I've renamed it to say "if there are no custom metadata definitions" which is a bit clearer.